tck/features/MatchAcceptance2.feature

```diff
@@ -0,0 +1,2023 @@
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2, count(*) AS c
+        ORDER BY c
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation and ORDER BY
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1 {id: 0}]->(),
+             ()-[:T2 {id: 1}]->()
+      """
+    When executing query:
+      """
+      MATCH (a)-[r]->(b)
+      WITH a, r, b, count(*) AS c
+        ORDER BY c
+      MATCH (a)-[r]->(b)
+      RETURN r AS rel
+        ORDER BY rel.id
+      """
+    Then the result should be, in order:
+      | rel           |
+      | [:T1 {id: 0}] |
+      | [:T2 {id: 1}] |
+    And no side effects
+
+  Scenario: Matching with LIMIT and optionally matching using an already bound relationship
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (:A)-[:T]->(:B)
+      """
+    When executing query:
+      """
+      MATCH ()-[r]->()
+      WITH r
+        LIMIT 1
+      OPTIONAL MATCH (a2)-[r]->(b2)
+      RETURN a2, r, b2
+      """
+    Then the result should be:
+      | a2   | r    | b2   |
+      | (:A) | [:T] | (:B) |
+    And no side effects
+
+  Scenario: Matching with LIMIT and optionally matching using an already bound relationship and node
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (:A)-[:T]->(:B)
+      """
+    When executing query:
+      """
+      MATCH (a1)-[r]->()
+      WITH r, a1
+        LIMIT 1
+      OPTIONAL MATCH (a1)-[r]->(b2)
+      RETURN a1, r, b2
+      """
+    Then the result should be:
+      | a1   | r    | b2   |
+      | (:A) | [:T] | (:B) |
+    And no side effects
+
+  Scenario: Matching with LIMIT, then matching again using an already bound relationship and node with additional predicate
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T]->()
+      """
+    When executing query:
+      """
+      MATCH (a1)-[r]->()
+      WITH r, a1
+        LIMIT 1
+      MATCH (a1:X)-[r]->(b2)
+      RETURN a1, r, b2
+      """
+    Then the result should be:
+      | a1 | r | b2 |
+    And no side effects
+
+  Scenario: Matching with LIMIT and predicates, then matching again using an already bound relationship and node with duplicate predicate
```

Comment:
"Matching with LIMIT and predicates, then matching again using an already bound relationship and node with duplicate predicate" -> "Matching with LIMIT and predicates, then matching again using a relationship and node that are both already bound along with a duplicate predicate"